GitUI/Theming/ThemeModule.cs

```diff
@@ -9,112 +12,101 @@ namespace GitUI.Theming
-            new Lazy<FormThemeEditor>(() => new FormThemeEditor(Controller));
-
-        private static ThemePersistence Persistence { get; } = new ThemePersistence();
-
-        private static readonly Lazy<Theme> DefaultThemeLazy =
-            new Lazy<Theme>(() => new DefaultTheme());
-
-        private static readonly Lazy<ThemeManager> ControllerLazy =
-            new Lazy<ThemeManager>(() => new ThemeManager(DefaultTheme));
-
-        private static readonly Lazy<SystemDialogDetector> SystemDialogDetectorLazy =
-            new Lazy<SystemDialogDetector>(() => new SystemDialogDetector());
-
-        private static ThemeDeployment Deployment =>
-            DeploymentLazy.Value;
-
-        public static FormThemeEditorController Controller =>
-            FormEditorControllerLazy.Value;
-
-        private static FormThemeEditor FormEditor =>
-            FormEditorLazy.Value;
-
-        private static Theme DefaultTheme =>
-            DefaultThemeLazy.Value;
-
-        private static ThemeManager ThemeManager =>
-            ControllerLazy.Value;
-
-        private static SystemDialogDetector SystemDialogDetector =>
-            SystemDialogDetectorLazy.Value;
+        public static ThemeSettings Settings { get; private set; } = ThemeSettings.Default;
 
         public static void Load()
         {
-            if (!TryInitialize())
-            {
-                // Load default theme to prevent mixing dark AppColors with bright SystemColors
-                // when AppSettings.UIThemeName points to dark theme
-                Controller.SetInitialTheme("win10default", useSystemVisualStyle: true);
-            }
+            new ThemeMigration(new ThemeRepository(new ThemePersistence())).Migrate();
+            Settings = TryLoadTheme();
+            ColorHelper.ThemeSettings = Settings;
+            ThemeFix.ThemeSettings = Settings;
+            Win32ThemeHooks.ThemeSettings = Settings;
         }
 
-        private static bool TryInitialize()
+        private static bool TryInstallHooks(Theme theme)
         {
+            Win32ThemeHooks.WindowCreated += Handle_WindowCreated;
+
             try
             {
-                Deployment.DeployThemesToUserDirectory();
+                Win32ThemeHooks.InstallHooks(theme, new SystemDialogDetector());
             }
             catch (Exception ex)
             {
-                // non mission-critical, proceed
-                Trace.WriteLine($"Failed to deploy schemes to user directory: {ex}");
+                Trace.WriteLine($"Failed to install Win32 theming hooks: {ex}");
+                Win32ThemeHooks.Uninstall();
+                return false;
             }
 
-            var invariantTheme = Controller.LoadInvariantTheme();
+            ResetGdiCaches();
+            return true;
+        }
+
+        private static ThemeSettings TryLoadTheme()
+        {
+            var repository = new ThemeRepository(new ThemePersistence());
```

Comment:
We creating a new repository on line:19. Can we use the same instance?

Reply:
ok